server/parameters/util/SubcommandParser.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2021 Vaticle
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.vaticle.typedb.core.server.parameters.util;
+
+import com.vaticle.typedb.core.common.exception.TypeDBException;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static com.vaticle.typedb.core.common.exception.ErrorMessage.Server.CLI_OPTION_MISSING_PREFIX;
+
+public abstract class SubcommandParser<RESULT> {
```

Comment:
The class is responsible for parsing the given subcommand according to the specified `tokens`. It is extracted from the `ServerSubcommand` class (which was called `Command`, prior to this PR).